### DISCUSSION TITLE
Error loading cache file for multiple delay equations

### DISCUSSION THREAD

**Discussion by SGeevers** [pymoca/pymoca#270] [Q&A] [answered]
When I have the following model:

```
model Example
    parameter Real t_delay = 0.1;
    output Real x;
    output Real y1;
    output Real y2;

equation
    der(x) = 1;
    y1 = delay(x, t_delay);
    y2 = delay(x, 0.2);

end Example;
```

and the following python script (in the same folder):

```python
import pathlib
import pymoca.backends.casadi.api

mdoel_folder = pathlib.Path(__file__).parent.resolve()
model_name = "Example"
compile_options = {
    'detect_aliases': True,
    'cache': True,
}
model = pymoca.backends.casadi.api.transfer_model(mdoel_folder, model_name, compile_options)
```

then everything goes well the first time I run the Python script, but the second time, when it loads the model from a cache file, I get the error `TypeError: unhashable type: 'list'`.

Any idea how to solve this issue? The issue does not occur when I replace the `0.2` in the model by `2 * t_delay` or when I replace `t_delay` in the equation by `0.1`.

**Comment by kentrutan** [comment]
Other than turning off cache? No. I'm not the CasADi expert here, so hoping someone else will chime in. 

What versions of Pymoca, Python and NumPy area you using? I get a different error  but it still fails.

**Comment by jackvreeken** [comment]
I also get a different error with NumPy 1.24.4 (and CasADi 3.5.5 or 3.6.3):

```
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.
```

I will take a look later tonight (possibly tomorrow), but either way would be good to know what NumPy + CasADi versions you are using.

**Comment by jackvreeken** [comment]
There's at least two small bugs in the caching for delayed states, will have a fix up later today.

**Reply by SGeevers** [reply]
Thanks for the quick replies! I used python 3.9, pymoca=0.9.0, casadi=3.5.5, numpy=1.22.4.

**Comment by kentrutan** [comment, marked as answer]
Fixed by #271.
